waku/v2/protocol/waku_store/waku_store.nim

```diff
@@ -614,6 +614,12 @@ proc queryWithAccounting*(ws: WakuStore, query: HistoryQuery, handler: QueryHand
     waku_store_errors.inc(labelValues = [dialFailure])
     return
 
+  #Check if the Peer has gone below the disconnect threshold
+  let peerConnectionStatus = ws.wakuSwap.getPeerConnectionStatus(peerOpt.get().peerId)
```

Comment:
is peerOpt.get().peerId always defined?

Reply:
Thanks. Though I Felt that will be covered by if `peerOpt.isNone()`